subsys/net/lib/lwm2m/lwm2m_rw_oma_tlv.c

```diff
@@ -926,7 +924,7 @@ int do_write_op_tlv(struct lwm2m_message *msg)
 		 * We need tlv.type to determine how to proceed.
 		 */
 		len = oma_tlv_get(&tlv, &msg->in, true);
-		if (len == 0) {
+		if (len < 0) {
```

Comment:
Empty data is ok?

Reply:
For TLV there's never a case where there is no data (i. e. 0 is returned) - either there is at least TLV header (and thus the return value is non-zero positive), or an error code is returned.